File: backend/src/main/java/com/qme/qmeapp/history/dto/HistoryEntryResponse.java
package com.qme.qmeapp.history.dto;

import lombok.Builder;
import lombok.Data;

import java.time.OffsetDateTime;
import java.util.UUID;

@Data
@Builder
public class HistoryEntryResponse {
    private UUID entryId;
    private UUID organizationId;
    private String organizationName;
    private String queueCode;
    private Integer queueNumber;
    private String status;       // SERVED, CANCELLED, SKIPPED
    private OffsetDateTime joinedAt;
    private OffsetDateTime completedAt; // servedAt, cancelledAt, or null
}

File: mobile/app/src/main/java/com/qme/mobile/data/model/response/HistoryEntryResponse.kt
package com.qme.mobile.data.model.response

data class HistoryEntryResponse(
    val entryId: String,
    val organizationId: String,
    val organizationName: String,
    val queueCode: String,
    val queueNumber: Int,
    val status: String,      // SERVED | CANCELLED | SKIPPED
    val joinedAt: String?,
    val completedAt: String?
)
